docs/design/coreclr/jit/Jit Stack Security.md

```diff
@@ -0,0 +1,64 @@
+# .NET Stack Security
+
+This document describes mechanisms in the .NET code generator to ensure stack security at runtime.
+
+## Background
+
+While .NET is primarily a type and memory safe "managed" programming platform, it also offers lower-level
+facilities to allow for interop with native code, as well as some constructs that cannot be proven safe.
+
+Use of these potentially "unsafe" constructs can threaten the integrity of the .NET runtime stack, enabling modification
+of key information on a stack frame, like the values of code and data addresses.
+
+The .NET code generator includes stack frame protection (aka Guard Stack or "GS") so that the integrity of the stack
```

Comment:
Wonder if it's worth mentioning how the same mechanism is called in other runtimes - Stack canaries?